### DISCUSSION TITLE
LIKE on List<String>

### DISCUSSION THREAD

**Discussion by jclohmann** [infinispan/infinispan#10116] [Q&A]
Hi, I am trying to query using the LIKE operator on a list of strings.

My java-type would be something like this.
```
class Author {
    String name;
    List<String> bookTitles;
}
```

Protostream generates a proto-definition like this:
```
message Author {
    optional string name = 1;
    repeated string bookTitles = 2;
}
```

When executing the query `FROM Author WHERE bookTitles like :param0` and param0 being "%infinispan%" i'm getting this error:
```
org.infinispan.HOTROD - ISPN004005: Error received from the server: org.infinispan.commons.CacheException: java.lang.ClassCastException
```

The documentation for `like` states `Checks that the left argument (which is expected to be a String) matches a wildcard pattern that follows the JPA rules.`.

Does that mean, that i won't be able to filter on a list? If so, are there any workarounds to get this running?

I am currently in the process of migrating from the deprecated QueryBuilder to the ickle-query language. Using the QueryBuilder I was able to do it like this:
```
queryBuilder.having("bookTitles").like("%infinispan%");
```

**Comment by jclohmann** [comment]
I found the ' : ' operator for fulltext search. This might be a solution for me.

But I'm unable to configure the indexing correctly.

This is my actual protofile:
```
// File name: xyz.proto
// Generated from : de.package.search.RemoteQueryInitializer

syntax = "proto2";



/**
 * @Indexed
 */
message Order {
   
   optional int64 id = 1;
   
   [some optional fields]
   
   /**
    * @Field(store = Store.YES, analyze=Analyze.YES, index=Index.YES, analyzer = @Analyzer(definition = "standard"))
    */
   optional string customerLastNameLowercase = 11;
   
   optional string customerType = 12;

   [some more optional fields]

}
```

When I'm trying to query the cache using this query in the infinispan-web-ui `FROM Order WHERE customerLastNameLowercase : '*kolbe*'` I get this response: `Error executing search The cache must be indexed in order to use full-text queries.`

I have no clue why the `@Indexed`-annotation won't enable the indexing.

Can someone please help me with this?

**Reply by StephenOTT** [reply]
I have been using with kotlin:

```
val listString = myList.joinToString { "\'$it\'" } // creates 'item1', 'item2', 'item3'
val query = "FROM book WHERE titles IN ($listString)"
```
